### ISSUE TITLE
Auth Blocking Function JWT Validation

### ISSUE THREAD

**Issue by tannerlinsley** [GoogleCloudPlatform/iap-gcip-web-toolkit#61]
When following the documentation for blocker functions [here](https://cloud.google.com/identity-platform/docs/blocking-functions#blocking_registration_or_sign-in) we noticed that there is quite a bit of internal JWT validation happening within this `beforeCreateHandler` abstraction:

```js
const gcipCloudFunctions = require('gcip-cloud-functions');
const authClient = new gcipCloudFunctions.Auth();

exports.beforeCreate = authClient.functions().beforeCreateHandler((user, context) => {
  // ... We can assume the token is valid at this point? How?
})
```

- Is it possible to perform this JWT verification on our own?
- What is the flow happening in this abstraction? Is there an outline or further details on this flow?

cc: @derekperkins

**Comment by prameshj** [comment]
Yes, you can assume that JWT verification has happened at this step. beforeCreateHandler takes in a `(user, context) => any` function and returns an http handler `(express.Request, express.Response) => Promise<void>`.

the `(user, context) => any` function is invoked only after performing a token verification via jsonwebtoken verify(). You could do the same by accessing the id token using `context.credential?.idToken?.`
